src/v/cloud_storage/partition_manifest.h

```diff
@@ -125,6 +128,15 @@ class partition_manifest final : public base_manifest {
     bool add(const key& key, const segment_meta& meta);
     bool add(const segment_name& name, const segment_meta& meta);
 
+    /// Remove segment from the manifest only if the metadata matches
+    bool remove(const key& key, const segment_meta& meta);
+    bool remove(const segment_name& key, const segment_meta& meta);
```

Comment:
looks like you don't use these methods in this PR, do you need them?

Reply:
removed it